// ng-image-slider/esm2020/lib/ng-image-slider.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { NgImageSliderComponent } from './ng-image-slider.component';
import { SliderCustomImageComponent } from './slider-custom-image/slider-custom-image.component';
import { SliderLightboxComponent } from './slider-lightbox/slider-lightbox.component';
import { NgImageSliderService } from './ng-image-slider.service';
import * as i0 from "@angular/core";
export class NgImageSliderModule {
}
/** @nocollapse */ NgImageSliderModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: NgImageSliderModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ NgImageSliderModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.0.4", ngImport: i0, type: NgImageSliderModule, declarations: [NgImageSliderComponent,
        SliderCustomImageComponent,
        SliderLightboxComponent], imports: [CommonModule], exports: [NgImageSliderComponent] });
/** @nocollapse */ NgImageSliderModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: NgImageSliderModule, providers: [
        NgImageSliderService
    ], imports: [CommonModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: NgImageSliderModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule
                    ],
                    declarations: [
                        NgImageSliderComponent,
                        SliderCustomImageComponent,
                        SliderLightboxComponent
                    ],
                    providers: [
                        NgImageSliderService
                    ],
                    exports: [NgImageSliderComponent]
                }]
        }] });
